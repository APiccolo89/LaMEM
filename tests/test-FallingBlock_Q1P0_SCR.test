EXEC            = ../bin/LaMEM -ParamFile ./input/FallingBlock_SCR.dat -vpt_element Q1P0  -AddRandomNoiseParticles 0 -time_end 1 -DisplayAverageQuantities
NPROCS          = 1
TEST_TYPE       = tolerance_diff_pl,1.0e-4
