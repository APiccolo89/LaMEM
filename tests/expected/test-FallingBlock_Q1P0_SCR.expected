# -------------------------------------------------------------------------- 
#                   Lithosphere and Mantle Evolution Model                   
#              Current Revision: 4812 - 2013-10-01 23:57:18		     
#  Modified items: LaMEM_Particles.c, Elements.c, Assembly_FDSTAG.c, fdstag/JacRes.c, 
#     Compiled: Date: Oct  9 2013 - Time: 23:02:54 - Mode:  	    
# -------------------------------------------------------------------------- 
 Parsing input file : ./input/FallingBlock_SCR.dat 
# Adding PetscOption: -vpt_element Q1P0 
# Adding PetscOption: -restart 0 
# Adding PetscOption: -save_breakpoints 0 
# Adding PetscOption: -scr_ksp_rtol 1.0e-4 
# Adding PetscOption: -scr_ksp_atol 1.0e-13 
# Adding PetscOption: -scr_pc_type bjacobi 
# Adding PetscOption: -vs_ksp_type fgmres 
# Adding PetscOption: -vs_pc_type fieldsplit 
# Adding PetscOption: -vs_fieldsplit_ksp_type fgmres 
# Adding PetscOption: -vs_fieldsplit_ksp_rtol 1.0e-2 
# Adding PetscOption: -vs_pc_fieldsplit_block_size 3 
# Adding PetscOption: -vs_pc_fieldsplit_type SYMMETRIC_MULTIPLICATIVE 
# LithColDepth[0] = 0 
# LithColDens[0] = 0 
# LithColDens[1] = 0 
# --------------------------------------------------------------------------
# Phase material parameters read from ./input/FallingBlock_SCR.dat: 
#    phase_1 (id=1), VISCOSITY, constant [0,0,0] = 1.000000e+03 
#    phase_1 (id=1), ELASTICITY, constant [0,1,0] = 5.000000e+10, 1.000000e+00 
#    phase_1 (id=1), DENSITY, temperature-dependent [0,2,0] = 2.000000e+00, 0.000000e+00, 0.000000e+00 
#    phase_1 (id=1), ENERGY, constant [0,3,0] = 3.000000e+00 1.050000e+03 0.000000e+00 0.000000e+00 
#    phase_0 (id=0), VISCOSITY, constant [1,0,0] = 1.000000e+00 
#    phase_0 (id=0), ELASTICITY, constant [1,1,0] = 5.000000e+10, 1.000000e+00 
#    phase_0 (id=0), DENSITY, temperature-dependent [1,2,0] = 1.000000e+00, 0.000000e+00, 0.000000e+00 
#    phase_0 (id=0), ENERGY, constant [1,3,0] = 3.000000e+00 1.050000e+03 0.000000e+00 0.000000e+00 
 Stokes solver             : Schur Complement Reduction
 Velocity solver           : User-defined (set options with prefix vs_)
 Note: Galerkin geometric multigrid is deactivated for user-defined solver 
 Element-type used         : Q1P0 ;  [Q1P0 Q2Pm1_global Q2Pm1_local Q1Q1 FDSTAG] (change with -vpt_element)
 Total # of cpu's          : 1 
 Total # of velocity dof's : 2187 
 Computing mean material per element using arithmetic averaging from integration points 
 Employing code in a Lagrangian manner 
 Gravity angle with z-axis =  0 
 Approximate Aspect ratio of cells [dx/dz, dy/dz] = [1 1] 
 Number of tracers/cell in [x,y,z]-direction = [2,2,2] 
 NumSurfaceNodes=1  nz=9 
 # Not employing nonlinear iterations 
 BC employed: BC.[Left Right; Front Back; Lower Upper]=[1 1; 1 1; 1 2] 
# Initial time step: 0.001 years 
 Input is assumed to be in non-dimensional units 
 Fine Grid parameters, nel=[8,8,8], cpu=[1,1,1] 
# Finished initializing Stokes matrices : 0.0106771 s 
# Not adding random noise to initial particle distribution. 
Initialized particles 
#rank=0, Attributing 4096 particles to elements took 0.00270295 s 
The total # of particles present is 4096 
 *** Using a falling block setup *** 
 Falling block coordinates:  [left,right]=[0.250000,0.750000]; [front,back]=[0.250000,0.750000]; [bottom,top]=[0.250000,0.750000] 
Saved initial particle distribution to directory ./InitialParticles 
The total memory used by integration points is 1.081344e+00 MB
The max memory used by integration points is 1.081344e+00 MB
The min memory used by integration points is 1.081344e+00 MB
#  Interpolating meshes and temperatures took 0.000931978 s 
#  Forming FEM stiffness matrix mx,my,mz=(9, 9, 9) ...   finished. [0.0217118 s]
 minDiagonalRatio=1 [Critical = 0.55; change with -CriticalDiagonalRatio]  
#  Forming stiffness matrixes took 0.021986 s 
SCR: Viscosity structure info: min(0)=1.000000e+00 : max(146)=1.000000e+03 : contrast=1.000000e+03 
  Residual norms for scr_ solve.
  0 KSP Residual norm 7.127495115274e-03 
  1 KSP Residual norm 3.140963402233e-04 
  2 KSP Residual norm 9.101625337127e-05 
  3 KSP Residual norm 4.652357591158e-05 
  4 KSP Residual norm 1.961881411908e-05 
  5 KSP Residual norm 1.174273428095e-05 
  6 KSP Residual norm 7.457261356158e-06 
  7 KSP Residual norm 3.827636516219e-06 
  8 KSP Residual norm 2.876521615521e-06 
  9 KSP Residual norm 1.851501630563e-06 
 10 KSP Residual norm 1.181282710252e-06 
 11 KSP Residual norm 9.022186995998e-07 
 12 KSP Residual norm 7.970562661764e-07 
 13 KSP Residual norm 6.510254270151e-07 
Linear solve converged due to CONVERGED_RTOL iterations 13
# Time: p solve = 1.21e-01 
# Time: u solve = 7.14e-03 
#  Finished Stokes solve after 0.14423 s 
------------------------------------------
Stokes solver summary: 
  divergence: 
    Div_min = 2.946662674504e-10 
    Div_max = 8.463781262007e-08 
    |Div|_2 = 6.650019070324e-07 
    |Div|_1 = 1.121700945792e-05 
  momentum: 
    mRes_min = 0.000000000000e+00 
    mRes_max = 4.258938681410e-08 
    |mRes|_2 = 1.990563375144e-07 
    |mRes|_1 = 3.418433163069e-06 
  velocity: 
    u_min   = -2.847990270257e-03 
    u_max   = 3.149073436144e-03 
    |u|_2   = 5.122018847437e-02 
    |u|_1   = 1.454034097697e+00 
  pressure: 
    p_min   = -6.017154755898e-01 
    p_max   = 5.234657140683e-01 
    |p|_2   = 7.488736068954e+00 
    |p|_1   = 1.465098513430e+02 
------------------------------------------
The total memory used by integration points is 1.081344e+00 MB
The max memory used by integration points is 1.081344e+00 MB
The min memory used by integration points is 1.081344e+00 MB
#  Solving for velocity took 0.17359 s 
#  Forming temperature stiffness matrix mx,my,mz=(9, 9, 9) ...  finished. 
#  Forming temperature stiffness matrixes took 0.00434804 s 
#  Solving for temperature took 0.00037694 s 
The total memory used by integration points is 1.081344e+00 MB
The max memory used by integration points is 1.081344e+00 MB
The min memory used by integration points is 1.081344e+00 MB
The total memory used by particles is 1.146880e+00 MB
The max memory used by particles is 1.146880e+00 MB
The min memory used by particles is 1.146880e+00 MB
#  Updating properties @ integration points and particles took 0.013463 s 
AdvectionMethod = 1 
  Starting particle advection 
  Finished particle advection 
# Volume-averaged quantities: 
# Txx = -0.000216447 Exx = 3.77194e-22 Volume = 1 Mu_xx=-2.86918e+17 
# Tyy = -0.000216457 Eyy = -4.06311e-21 Volume = 1 Mu_yy=2.66368e+16 
# Tzz = 0.000428513 Ezz = -5.90805e-20 Volume = 1 Mu_zz=-3.62651e+15 
# Txz = 1.27538e-08 Exz = 5.09265e-09 Volume = 1 Mu_xz=1.25218 
# Txy = -1.71514e-09 Exy = 3.78506e-10 Volume = 1 Mu_xy=-2.26568 
# Tyz = 7.88569e-09 Eyz = -1.1429e-09 Volume = 1 Mu_yz=-3.44987 
Minimum z-level phase 1 0.281247 
# Writing output to existing directory Timestep_000000 
# Saved MATLAB output in file Timestep_000000/FB_test.0.1000000.out 
# Saved Paraview VTS Velocity output in file FB_test_velocity_step000000-mesh.pvts 
# Saved Paraview VTS Temperature output in file FB_test_temperature_step000000-mesh.pvts 
# Saved Paraview VTS Surface Topography output in file FB_test_Surface_topography_step000000-mesh.pvts 
# Saved Paraview VTS Bottom Topography output in file FB_test_Bottom_topography_step000000-mesh.pvts 
# Saved Paraview VTS point-wise quadrature output in file FB_test_quadrature_step000000-mesh.pvts 
 Maximum velocity  0.00314907    
 Time = 0.001, dt=1e-05   
# Finished timestep 0 out of 1 in 0.211076 s
  
# Total time required: 0.24094 s 
#PETSc Option Table entries:
-AddRandomNoiseParticles 0
-DisplayAverageQuantities
-ParamFile ./input/FallingBlock_SCR.dat
-options_left
-restart 0
-save_breakpoints 0
-scr_ksp_atol 1.0e-13
-scr_ksp_converged_reason
-scr_ksp_max_it 1000
-scr_ksp_monitor
-scr_ksp_rtol 1.0e-4
-scr_ksp_type fgmres
-scr_pc_type bjacobi
-time_end 1
-vpt_element Q1P0
-vs_fieldsplit_ksp_rtol 1.0e-2
-vs_fieldsplit_ksp_type fgmres
-vs_ksp_type fgmres
-vs_pc_fieldsplit_block_size 3
-vs_pc_fieldsplit_type SYMMETRIC_MULTIPLICATIVE
-vs_pc_type fieldsplit
#End of PETSc Option Table entries
There are no unused options.
